src/lib-components/navigation/TablePaginator.vue

```diff
@@ -0,0 +1,146 @@
+<script setup lang="ts">
+import { Pagination } from "@/composables/nav"
+import { debounceFn } from "@/entry"
+import { NumberInput, Select } from "@/lib-components"
+import { computed } from "vue"
+
+const props = defineProps<{
+  pageOptions?: { label: string; value: number }[]
+}>()
+
+const pagination = defineModel<Pagination>({ required: true })
+
+const page = computed({
+  get: () => pagination.value.page,
+  set: (v: number) => {
+    pagination.value = {
+      ...pagination.value,
+      page: v,
+    }
+  },
+})
+
+const perPage = computed({
+  get: () => pagination.value.perPage,
+  set: (v: number) => {
+    pagination.value = {
+      ...pagination.value,
+      perPage: v,
+    }
+  },
+})
+
+const pageSelectOpts = computed(() => {
+  if (props.pageOptions) {
+    return props.pageOptions
+  }
+
+  return [
+    { label: "5 per page", value: 5 },
+    { label: "10 per page", value: 10 },
+    { label: "20 per page", value: 20 },
+    { label: "50 per page", value: 50 },
```

Comment:
Looks like a good first pass. I'm failing to think of many situations where, say, 100 or even 50 results would be helpful. Whenever I've used those larger per page numbers in apps, my impression its something like (1) I don't know what to search for (e.g., unfamiliar with the data, search sucks, etc.) or (2) I want to see 1 or 2 fields on every line item (e.g. credit card statement).